app/src/main/java/de/danoeh/antennapod/fragment/QueueFragment.java

```diff
@@ -365,6 +390,21 @@ public void onConfirmButtonPressed(
         }
     }
 
+    /**
+     * This method is called if the user clicks on a sort order menu item.
+     *
+     * If the queue is in keep sorted mode, the new sort order is stored in the preferences and
+     * the queue is sorted. Otherwise the queue is just sorted.
+     *
+     * @param sortOrder New sort order.
+     */
+    private void setSortOrder(SortOrder sortOrder) {
+        boolean keepSorted = UserPreferences.isQueueKeepSorted();
+        if (keepSorted) {
+            UserPreferences.setQueueKeepSortedOrder(sortOrder);
```

Comment:
This does not need to be conditional anymore. If the queue just saves the order every time, it would support first sorting and then checking "keep sorted".

Reply:
You are right, that's a good idea as we moved down the deep sorted item. I will change that.